hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -366,22 +484,26 @@ private void generateSnapshotDiffReport(final String jobId,
 
   private void addToObjectIdMap(Table<String, ? extends WithObjectID> fsTable,
                                 Table<String, ? extends WithObjectID> tsTable,
-                                Set<String> deltaFiles,
+                                Pair<Boolean, Set<String>>
+                                        isNativeRocksToolsLoadedDeltaFilesPair,
                                 PersistentMap<byte[], byte[]> oldObjIdToKeyMap,
                                 PersistentMap<byte[], byte[]> newObjIdToKeyMap,
                                 PersistentSet<byte[]> objectIDsToCheck,
                                 Map<String, String> tablePrefixes)
-      throws IOException {
+      throws IOException, NativeLibraryNotLoadedException {
+    Set<String> deltaFiles = isNativeRocksToolsLoadedDeltaFilesPair.getRight();
+    boolean nativeRocksToolsLoaded =
+            isNativeRocksToolsLoadedDeltaFilesPair.getLeft();
+    boolean isDirectoryTable =
+        fsTable.getName().equals(OmMetadataManagerImpl.DIRECTORY_TABLE);
```

Comment:
nit: this can be move after line no 502.

Reply:
done